cli/command/stack/cmd.go

```diff
@@ -15,19 +18,24 @@ func NewStackCommand(dockerCli command.Cli) *cobra.Command {
 		RunE:        command.ShowHelp(dockerCli.Err()),
 		Annotations: map[string]string{"version": "1.25"},
 	}
-	cmd.AddCommand(
-		newDeployCommand(dockerCli),
-		newListCommand(dockerCli),
-		newRemoveCommand(dockerCli),
-		newServicesCommand(dockerCli),
-		newPsCommand(dockerCli),
-	)
+	switch orchestrator.GetOrchestrator(dockerCli) {
+	case orchestrator.Kubernetes:
+		kubernetes.AddStackCommands(cmd, dockerCli)
+	case orchestrator.Swarm:
+		swarm.AddStackCommands(cmd, dockerCli)
+	}
 	return cmd
 }
 
 // NewTopLevelDeployCommand returns a command for `docker deploy`
 func NewTopLevelDeployCommand(dockerCli command.Cli) *cobra.Command {
-	cmd := newDeployCommand(dockerCli)
+	var cmd *cobra.Command
+	switch orchestrator.GetOrchestrator(dockerCli) {
+	case orchestrator.Kubernetes:
+		cmd = kubernetes.NewTopLevelDeployCommand(dockerCli)
+	case orchestrator.Swarm:
+		cmd = swarm.NewTopLevelDeployCommand(dockerCli)
+	}
 	// Remove the aliases at the top level
 	cmd.Aliases = []string{}
 	cmd.Annotations = map[string]string{"experimental": "", "version": "1.25"}
```

Comment:
"version_kubernetes": "v1beta1"?

Reply:
@AkihiroSuda depends on what you wanna check here : the version of kubernetes, the version of the "k8s" native api or the version of our api extension — or all of those 😝.

Reply:
I think the version of the stack API extension is correct to check here. (And the stack API extension may provide info for minimum supported k8s version and so on)